bin/generate_missing_mappings

```diff
@@ -0,0 +1,328 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require_relative "../config/environment"
+require_relative "../config/cli"
+require "byebug"
+require "qdrant"
+require "bigdecimal"
+require "openai"
+require "csv"
+require "yaml"
+
+class TaxonomyMapper
+  MAX_RETRIES = 3
+  QDRANT_PORT = 6333
+  EMBEDDING_MODEL = "text-embedding-3-small"
+  MAPPING_GRADER_GPT_MODEL = "gpt-4"
+  LATEST_SHOPIFY_VERSION = "shopify/#{File.read("VERSION").strip}"
+
+  def initialize
+    @cli = CLI.new
+    @openai_client = create_openai_client
+    @qdrant_client = create_qdrant_client
+  end
+
+  def run
+    mappings_json_data = @cli.parse_json("dist/en/integrations/all_mappings.json")
+    shopify_categories_missing_mapping_groups = find_unmapped_shopify_categories(mappings_json_data)
+    return if shopify_categories_missing_mapping_groups.empty?
+
+    shopify_categories_missing_mapping_groups.each do |missing_mapping_group|
+      generate_missing_mappings(missing_mapping_group)
+    end
+  end
+
+  private
+
+  def create_openai_client
+    OpenAI::Client.new(
+      access_token: ENV["OPENAI_API_TOKEN"],
+      uri_base: "https://openai-proxy.shopify.ai/v1",
+      request_timeout: 10,
+    )
+  end
+
+  def create_qdrant_client
+    ensure_qdrant_server_running
+    Qdrant::Client.new(url: "http://localhost:#{QDRANT_PORT}")
+  end
+
+  def ensure_qdrant_server_running
+    return if system("lsof -i:#{QDRANT_PORT}", out: "/dev/null")
+
+    command = "podman run -p #{QDRANT_PORT}:#{QDRANT_PORT} qdrant/qdrant"
+    pid = Process.spawn(command, out: "/dev/null", err: "/dev/null")
+    Process.detach(pid)
+    puts "Started Qdrant server in the background with PID #{pid}."
+  end
+
+  def find_unmapped_shopify_categories(mappings_json_data)
+    shopify_categories_lack_mappings = []
```

Comment:
Been thinking about this for a few of our commands if we'd be better served by leveraging our SQL vs the dist jsons 🤔 Not sure if that'd apply here or not though